### DISCUSSION TITLE
how to know flask_socketio is connected

### DISCUSSION THREAD

**Discussion by 6273parastu** [miguelgrinberg/Flask-SocketIO#1898] [Q&A]
hi, i need to know connection socketio is alive or is connected when i use in `start_background_task`, for e.x:
`@socketio.on('connect')`
`@authenticated_only`
`def on_connect():`
    `      connection_id = request.sid`
    `     socketio.start_background_task(target=background_thread_from_stream, app=current_app._get_current_object(), connection_id=connection_id)`

`def background_thread_from_stream(self, app, connection_id):`
        `with app.app_context():`
            `while connection_id is connected:`
                   `do_somtitng()`

**Comment by miguelgrinberg** [comment]
When a client disconnects the `disconnect` handler is invoked. You can keep track of your users in your application, adding users in the connect handler, and removing them in the disconnect handler.
